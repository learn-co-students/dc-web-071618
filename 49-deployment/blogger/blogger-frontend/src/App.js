import React, { Component } from "react";
import { connect } from "react-redux";
import "./App.css";
import { Switch, Route, withRouter } from "react-router-dom";
import NavBar from "./components/NavBar";
import Login from "./components/Login";
import Home from "./components/Home";
import UserPage from "./components/UserPage";
import UserList from "./components/UserList";
import PostPage from "./components/PostPage";
import PostList from "./components/PostList";
import { fetchUsers, fetchPosts } from "./redux/actions";

class App extends Component {
  componentDidMount() {
    this.props.fetchUsers();
    this.props.fetchPosts();
  }

  render() {
    return (
      <div className="App">
        <NavBar
          color="blue"
          icon="pencil"
          title="Postr"
          subtitle="blogging since 2003"
        />
        <Switch>
          <Route path="/login" component={Login} />
          <Route path="/users/:userId" component={UserPage} />
          <Route path="/users" component={UserList} />
          <Route path="/posts/:postId" component={PostPage} />
          <Route path="/posts" component={PostList} />
          <Route path="/*" component={Home} />
        </Switch>
      </div>
    );
  }
}

export default withRouter(
  connect(
    null,
    { fetchUsers, fetchPosts }
  )(App)
);
