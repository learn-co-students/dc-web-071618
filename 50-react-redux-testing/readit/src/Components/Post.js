import React from "react";

const Post = ({ post }) => {
  return (
    <div className="postComponent">
      <a href={post.url}>
        <h3>{post.title}</h3>
      </a>
      <p> posted_by: {post.author}</p>
    </div>
  );
};

export default Post;
