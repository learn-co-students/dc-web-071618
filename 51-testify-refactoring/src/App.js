import './App.css';

class GildedRose {
    constructor(sellIn, quality, name) {
        this.name = name;
        this.sellIn = sellIn;
        this.quality = quality;
    }

    tick () {
        if (this.name === 'normal') { return this.normalTick(); }
        if (this.name === 'Aged Brie') { return this.brieTick(); }
        if (this.name === 'Sulfuras, Hand of Ragnaros') {
            return this.sulfurasTick();
        }
        if (this.name === 'Backstage passes to a TAFKAL80ETC concert') {
            return this.backstageTick();
        }
   }

    normalTick () {
        this.sellIn = this.sellIn - 1;
        this.quality = this.quality - 1;
        if (this.sellIn <= 0) { this.quality = this.quality - 1; }
        if (this.quality < 0) { this.quality = 0; }
    }

    brieTick() {
        this.sellIn = this.sellIn - 1;
        this.quality = this.quality + 1;
        if (this.sellIn <= 0) { this.quality = this.quality + 1; }
        if (this.quality > 50) { this.quality = 50; }
    }

    sulfurasTick() { }

    backstageTick() {
        if (this.sellIn > 10) { this.quality = this.quality + 1; }
        if (this.sellIn > 5 && this.sellIn <= 10) {
            this.quality = this.quality + 2;
        } else if (this.sellIn <= 5) {
            this.quality = this.quality + 3;
        }
        if (this.sellIn <= 0) { this.quality = 0; }
        if (this.quality > 50) { this.quality = 50; }
        this.sellIn = this.sellIn - 1;
    }

}

export default GildedRose;
