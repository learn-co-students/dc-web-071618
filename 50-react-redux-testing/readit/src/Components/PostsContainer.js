import React from "react";
import Post from "./Post";
import { connect } from "react-redux";

const PostsContainer = ({ posts }) => {
  ///EXPECT TO MATCH SNAPSHOT
  return (
    <div className="postsContainer">
      {posts ? posts.map(p => <Post post={p.data} key={p.data.id} />) : null}
      <p>hello</p>
    </div>
  );
};

const mapStateToProps = state => {
  return {
    posts: state.posts
  };
};

export default connect(mapStateToProps)(PostsContainer);
export { PostsContainer };
