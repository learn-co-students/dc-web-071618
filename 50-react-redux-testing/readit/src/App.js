import React, { Component } from "react";
import { connect } from "react-redux";
import * as actions from "./actions";
import data from "./data/data";

import PostsContainer from "./Components/PostsContainer";

import "./App.css";

class App extends Component {
  componentDidMount() {
    return this.props.loadData
      ? this.props.loadData(data)
      : console.log("shallow, just 4 testing!");
  }

  render() {
    return (
      <div className="App">
        <h1>READIT</h1>
        <PostsContainer />
      </div>
    );
  }
}

export default connect(
  null,
  actions
)(App);

export { App };
