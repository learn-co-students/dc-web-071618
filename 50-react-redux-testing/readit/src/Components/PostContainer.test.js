import React from "react";
import { shallow } from "enzyme";

import Post from "./Post";
import { PostsContainer } from "./PostsContainer";
import App from "../App";

//test for rendering

describe("PostsContaier", () => {
  it("Renders correctly", () => {
    let wrapper = shallow(<PostsContainer />);
    expect(wrapper.find(".postsContainer").length).toEqual(1);
  });
  it("matches its snapshot when it receives posts as a prop", () => {
    let posts = [
      { data: { id: 1, title: "post1", author: "me" } },
      { data: { id: 2, title: "post2", author: "you" } },
      { data: { id: 3, title: "post3", author: "bob" } }
    ];
    let wrapper = shallow(<PostsContainer posts={posts} />);
    expect(wrapper).toMatchSnapshot();
  });
  it("matches its snapshot when it receives no props", () => {
    let wrapper = shallow(<PostsContainer />);
    expect(wrapper).toMatchSnapshot();
  });
});
