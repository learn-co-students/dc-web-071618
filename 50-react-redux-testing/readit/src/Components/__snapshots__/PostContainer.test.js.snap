// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostsContaier matches its snapshot when it receives no props 1`] = `
<div
  className="postsContainer"
>
  <p>
    hello
  </p>
</div>
`;

exports[`PostsContaier matches its snapshot when it receives posts as a prop 1`] = `
<div
  className="postsContainer"
>
  <Post
    key="1"
    post={
      Object {
        "author": "me",
        "id": 1,
        "title": "post1",
      }
    }
  />
  <Post
    key="2"
    post={
      Object {
        "author": "you",
        "id": 2,
        "title": "post2",
      }
    }
  />
  <Post
    key="3"
    post={
      Object {
        "author": "bob",
        "id": 3,
        "title": "post3",
      }
    }
  />
  <p>
    hello
  </p>
</div>
`;
